api/configserver/save.go

```diff
@@ -106,11 +116,24 @@ func (s *Server) SaveConfig(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	session.Info("saving")
-
 	pipelineName := rata.Param(r, "pipeline_name")
 	teamName := rata.Param(r, "team_name")
 
+	if checkCredentials {
+		variables := s.variablesFactory.NewVariables(teamName, pipelineName)
+		_, isNoop := variables.(*noop.Noop)
```

Comment:
Hmm I don't think we need this special case. I could see a user still wanting to check that all vars resolve, and expect this to fail, since in reality their builds will fail. That way the user can know that the vars aren't satisfied by a remote credential manager, so they need to provide them locally with `-v`/`-y`/`-l`.

Reply:
Ah yes, that makes sense.  The `Noop` special case is now gone.